### PR TITLE
Set up server side rendering with Vite and a working docker image + compose

### PR DESCRIPTION
**Pull request by dawnofmidnight**
At the moment, the application is rendered completely client side, which loads a LOT of JS into the browser. Instead, the app is now rendered server side with the option to do it client side as well, which will make it work with Tauri.

### CODE REVIEW

File: backend/src/main.py [hunk 1]
```diff
@@ -13,4 +13,11 @@
 @app.get("/")
 async def home(request: Request):
     logger.info(f"GET request to endpoint / from client {request.client.host}")
+    logger.trace(f"GET request to endpoint / from client {request.client.host}")
```
Comment (Evorage0) [inline backend/src/main.py:16]: kowalski, analysis

File: frontend/.eslintrc.cjs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    indent: ["error", 2],
+    quotes: ["error", "double"],
+    semi: ["error", "never"],
+    curly: ["error", "all"],
+    "dot-location": ["error", "property"],
+    "no-floating-decimal": "error",
+    "no-multi-spaces": "error",
+    "no-trailing-spaces": "error",
+    "jsx-quotes": ["error", "prefer-double"],
+    // "@typescript-eslint/no-var-requires": "off"
```
Comment (dawnofmidnight) [inline frontend/.eslintrc.cjs:36]: Remove this line

File: frontend/src/lib/requests/utils/logger.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  patch(url: string | undefined): void {
+    console.log(`${chalk.green(now())} | ${chalk.hex("#91fff3")("PATCH")}    | endpoint: ${url}`)
+  }
+
+  delete(url: string | undefined): void {
+    console.log(`${chalk.green(now())} | ${chalk.hex("#91fff3")("DELETE")}   | endpoint: ${url}`)
+  }
+
+  log(method: string | undefined, url: string | undefined): void {
+    console.log(`${chalk.green(now())} | ${logMethod(method)} | endpoint: ${url}`)
```
Comment (dawnofmidnight) [inline frontend/src/lib/requests/utils/logger.ts:35]: See if you can invoke it's own class methods instead.

File: frontend/src/lib/utils/logger.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  error(message: string): void {
+    console.log(`${chalk.green(now())} | ${chalk.red("ERROR")}    | ${message}`)
+  }
+
+  critical(message: string): void {
+    console.log(`${chalk.green(now())} | ${chalk.white.bgRed("CRITCAL")}  | ${message}`)
+  }
+
+  log(type: string, message: string): void {
+    console.log(`${chalk.green(now())} | ${logType(type)} | ${message}`)
```
Comment (dawnofmidnight) [inline frontend/src/lib/utils/logger.ts:48]: Same here

**Pull request closed without merging**